WI #647: Adds `Amortization Class` column (#747)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/columns/amortizationclass.md

> # Amortization Class

Reviewer: There are a few things about this that seem problematic to me
- We're really talking about pre-payment here and the process of burning down pre-payments.
- When you pre-pay a contract (be it SaaS or Cloud or whatever), and burn that down (possibly very unevenly) over a term, are those charges considered 'Amortized Charges'? To me, it doesn't seem that way.
- This says allows nulls False but the normative statements says this should be null when its not Principal or the Amortized Charge.

The original fix may need to happen in EffectiveCost to avoid talking about amortized cost. 

Would something like ChargeCoverageCategory with values like 
- Pre-payment Charge: Billed > 0, effective = 0
- Burndown Charge: Billed=0, effective>0
- Standard Charge: Billed = effective work? 

If too late to take these suggestions, respond here as I can't attend the calls.

Reviewer 2: We discussed this in TF1 last week. We determined that expanding to other scenarios doesn't make sense because those scenarios require different handling. That would be overloading a single column with multiple purposes and indicating which applies would be difficult. We decided to keep the initial scope of only handling amortization.

Reviewer 2: FYI - Discussed again in TF1 on Mar 18

Reviewer 2: FYI - Added more context to the supporting content to explain why we can't cover all double counting scenarios in one column. Specifically, commitment discount purchases can be both credit burndown targets and original amortization charges.

> Amortization Class indicates how a row in a [*FOCUS dataset*](#glossary:FOCUS-dataset) relates to [*amortization*](#glossary:amortization). Amortization Class is used to identify and distinguish between the charges before and after *amortization* is applied. Amortization Class is commonly used to quantify cost savings accurately by excluding charges that can be double-counted when aggregated List Cost or Contracted Cost.

Reviewer 3: is there verbiage missing here? this last sentence does not read fluidly to me, as if it it is missing some important punctuation or another phrase. 

"Amortization Class is commonly used to quantify cost savings accurately by excluding charges that can be double-counted when aggregated List Cost or Contracted Cost."

Specifically this clause, 'excluding charges that can be double-counted when aggregated List Cost or Contracted Cost'

Author: There are a few of these proofreading issues I will take care of today.

Reviewer 2: This seems simple...
```suggestion
Amortization Class indicates how a row in a [*FOCUS dataset*](#glossary:FOCUS-dataset) relates to [*amortization*](#glossary:amortization). Amortization Class is used to identify and distinguish between the charges before and after *amortization* is applied. Amortization Class is commonly used to quantify cost savings accurately by excluding charges that can be double-counted when aggregating List Cost or Contracted Cost.
```

> Amortization Class indicates how a row in a [*FOCUS dataset*](#glossary:FOCUS-dataset) relates to [*amortization*](#glossary:amortization). Amortization Class is used to identify and distinguish between the charges before and after *amortization* is applied. Amortization Class is commonly used to quantify cost savings accurately by excluding charges that can be double-counted when aggregating List Cost or Contracted Cost.

Reviewer 2: ```suggestion
Amortization Class indicates how a row in a [*FOCUS dataset*](#glossary:FOCUS-dataset) relates to [*amortization*](#glossary:amortization). Amortization Class is used to identify and distinguish between the charges before and after *amortization* is applied. Amortization Class is commonly used to quantify cost savings accurately by excluding the original principal charges that are amortized over time, which could otherwise be double-counted when aggregating List Cost or Contracted Cost.
```

The AmortizationClass column adheres to the following requirements:

* AmortizationClass column MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports *amortization*.
* AmortizationClass MUST be of type String.
* AmortizationClass MUST be "Principal" when the row represents an original charge that is amortized across multiple other rows.

> * AmortizationClass MUST be "Amortized Charge" when the row represents an result of *amortization* from a previous charge.

Reviewer 3: nit - 'an result' should be 'a result'

Reviewer 2: ```suggestion
* AmortizationClass MUST be "Amortized Charge" when the row represents a result of *amortization* from a previous charge.
```

> * AmortizationClass MUST be "Amortized Charge" when the row represents a result of *amortization* from a previous charge.

Reviewer 4: For AWS, EC2 Capacity Block for ML has the following behavior --

For the upfront charge, billed cost >0, effective cost =0
For the hourly charge, billed cost >0 (if the instance is using a paid OS, the billed cost represents the OS licensing fee), effective cost >0 (OS licensing fee + amortization of upfront Capacity Block for ML payment)

Need to clarify what does "represents a result of" in the proposal.

`* AmortizationClass MUST be "Amortized Charge" when the row represents a result of *amortization* from a previous charge.
`
Update to
`* AmortizationClass MUST be "Amortized Charge" when the row includes *amortization* from a previous charge.
`

Reviewer 2: @letian-wang-92 FYI - you can make a suggestion using the "+/-" file icon in the toolbar when leaving a comment inline.
```suggestion
* AmortizationClass MUST be "Amortized Charge" when the row includes *amortization* from a previous charge.
```

* AmortizationClass MUST be null when it is not the original, principal charge or the effective, amortized charge after *amortization* is applied.

## Column ID

AmortizationClass

## Display Name

Amortization Class

## Description

Indicates how a row relates to amortization to identify and distinguish between the charges before and after amortization is applied.

## Content Constraints

| Constraint    | Value          |
| :------------ | :------------- |
| Column type   | Dimension      |
| Feature level | Conditional    |

> | Allows nulls  | False          |

Reviewer 2: ```suggestion
| Allows nulls  | True           |
```

| Data type     | String         |
| Value format  | Allowed values |

Allowed values:

| Value            | Description                                                                                                                       |
| :--------------- | :-------------------------------------------------------------------------------------------------------------------------------- |
| Principal        | Original charge that is amortized over time across multiple rows.                                                                 |

> | Amortized Charge | Effective charge after the principal amount is amortized over time based on the associated ChargePeriodStart and ChargePeriodEnd. |

Reviewer 5: I think it makes sense to approach this in iteration, but I do think we should better specify "Amortized Charge" so it doesn't roadblock us later on.  AFAIK, a post-contract commitment discount is an amortized charge is an example of [straight-line amortization](https://en.wikipedia.org/wiki/Amortization_schedule), or equal payments over the term.

In the case of a pre-paid contract (i.e. Datadog) of funds over a term, I think it's valid to argue that the amortization schedule is reliant on the user's usage activity, so this is another form of amortization.

As we start to mix SaaS datasets, I think we should differentiate between different principals and amortization types with more specific values like (these are bad names):
- Principal --> Post-Contract Principal
- Amortized Charge --> Post-Contract Amortized Charge

so we can then include and differentiate values like (these are bad names):
- Contract Principal (i.e. SaaS Upfront Payment)
- Contract Amortized Charge (i.e. SaaS usage)

Reviewer 2: I don't quite understand the scenario. We might need to talk thru it. But regardless, if it's about amortization, then there's the original charge with BilledCost that gets amortized and then there are the "target" charges that get the amortized amounts with EffectiveCost. For the purpose of identifying how the rows apply amortization, do we care whether it's pre or post? 🤔 I can't say definitively since I don't completely understand the scenario, but if we're creating this to prevent double counting the original and target charges, then we should be able to use the same values for both.

Reviewer 6: I think it might be good to show more examples if possible so I understand what Chris is saying from the SaaS side and agree with the iteration but how would it work from the public cloud side as an example?

Reviewer 2: As discussed in TF1, this suggestion wouldn't work because it's merging two concepts into one column. One is the type of purchase and the other is how amortization is applied. This would need to be split out into separate columns:

| Scenario | PurchaseType | AmortizationClass |
|-----------|-----------|-----------|
| Commitment discount purchase | Commitment Discount | Principal |
| Commitment discount usage | Commitment Discount | Amortized Charge |
| Amortized Marketplace purchase | Marketplace | Principal |
| Amortized Marketplace usage | Marketplace | Amortized Charge |
| Amortized credit purchase | Credit | Principal |
| Amortized credit usage | Credit | Amortized Charge |

The only reason we need additional values is if there's another scenario where a row is impacted by amortization in a different way than the original charge or the allocated breakdown.

Reviewer 2: More importantly for this comment, I researched amortization and this credit burndown scenario doesn't seem to apply. From what I'm seeing on accounting websites, amortization always has a structured, consistent payment schedule called an "amortization schedule". This is usually monthly, daily, or hourly, like for commitment discounts. While we apply amortization differently than a loan, the amortization schedule is effectively the same where the original purchase ("principal" in accounting terms) is broken down over a structured, consistent usage schedule. This isn't applicable in the credit burndown scenario. While you could use BilledCost for the credit payment and EffectiveCost for the credit usage, it is not amortization and therefore would not have an AmortizationClass value.

I think we need to provide more context on amortization and the use of BilledCost and EffectiveCost in non-amortized scenarios like the credit burndown scenario. We should look at that for 1.3.

Reviewer: The last two places I worked at both had accounting people who were debating if things like cloud costs, including reservations, could be capitalized. I see the same arguments online. I don't think we're going to get an easy/clear answer on if we should consider this amortization.

For commitments that are charged hourly or daily even though they are for a year or longer (which I believe is the case for all 3 big CSPs), how does this model the charges? Do you have any sample data for the common commitment scenarios? Does amortization apply to those no up-front commitments?
